src/cli/cli_dataset.cpp

```diff
@@ -578,28 +578,29 @@ otError Dataset::ProcessCommand(const ComponentMapper &aMapper, Arg aArgs[])
     return error;
 }
 
-otError Dataset::Print(otOperationalDatasetTlvs &aDatasetTlvs)
+otError Dataset::Print(otOperationalDatasetTlvs &aDatasetTlvs, bool aNonsensitiveOnly)
 {
     struct ComponentTitle
     {
-        const char *mTitle; // Title to output.
-        const char *mName;  // To use with `LookupMapper()`.
+        const char *mTitle;       // Title to output.
+        const char *mName;        // To use with `LookupMapper()`.
+        const bool  mIsSensitive; // Whether the field is sensitive.
```

Comment:
Remove `const`.
```suggestion
      bool  mIsSensitive; // Whether the field is sensitive.
```

The whole `ComponentTitle` array is defined as `const`, so all its member variables are treated as `const`.

Note that `const char *mTitle` is different; it means `mTitle` is a pointer to a `char` buffer, and the content it points to is `const`. This does not mean the `mTitle` pointer itself is `const` (i.e., the `mTitle` variable, which is a pointer, can be changed to point elsewhere). The location of the `const` specifier matters. For example, `char *const mTitle` means it is a `const` pointer (the pointer itself cannot be changed to point somewhere else), but the `char` value it points to can be changed.

Reply:
Done.